libraries/aws_internet_gateways.rb

```diff
@@ -0,0 +1,64 @@
+# frozen_string_literal: true
+
+require 'aws_backend'
+
+class AwsInternetGateways < AwsResourceBase
+  name 'aws_internet_gateways'
+  desc 'Verifies settings for a collection of AWS internet gateways.'
+
+  example "
+    describe aws_internet_gateways do
+      its('count') { should eq 3 }
+    end
+
+    describe aws_internet_gateways.where(name: 'my-gateway') do
+      it { should exist }
+    end
+  "
+
+  attr_reader :table
+
+  FilterTable.create
+             .register_column(:ids, field: :id)
+             .register_column(:names, field: :name)
+             .register_column(:tags, field: :tags)
+             .register_column(:vpc_ids, field: :vpc_id)
+             .register_column(:owner_ids, field: :owner_id)
+             .register_column(:attachment_states, field: :attachment_state)
+             .install_filter_methods_on_resource(self, :table)
+
+  def initialize(opts = {})
+    super(opts)
+    validate_parameters
+    @table = fetch_data
+  end
+
+  def fetch_data
+    igw_rows = []
+    query_params = {}
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.compute_client.describe_internet_gateways(query_params)
+      end
+      return [] if !@api_response || @api_response.empty?
+
+      @api_response.internet_gateways.each do |igw|
+        unless igw[:tags].empty?
+          name_tag = igw[:tags].select { |tag| tag[:key] == 'Name' }
+          name = name_tag.first[:value]
+        end
```

Comment:
This could perhaps be reduced to:
```
name = igw[:tags].select { |tag| tag[:key] == 'Name' }.first[:value] unless igw[:tags].empty?
```

Reply:
Actually, there is a chance that the `select` method can return an empty array, and it requires a check on `name_tag` before returning the `value` of the first element.
Here is what I'm thinking:
```ruby
        unless igw[:tags].empty?
          name_tag = igw[:tags].select { |tag| tag[:key] == 'Name' }
          name = name_tag.first[:value] unless name_tag.empty?
        end
```

Reply:
This should hopefully work as expected
```
igw[:tags].select{ |tag| tag[:key] == 'Name' }.first&.dig(:value)
```
The `&.dig(:value)` part will just return nil if the key doesn't exist.

Reply:
In fact, this could probably be used in the row assignment eg.
```
name: igw[:tags].select{ |tag| tag[:key] == 'Name' }.first&.dig(:value)
```

Reply:
Thanks for that @rmoles.
Waiting for GH-209 to be resolved.